aws/resource_aws_elasticsearch_domain.go

```diff
@@ -169,6 +169,38 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 					},
 				},
 			},
+			"log_publishing_options": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"log_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+								value := v.(string)
+								validLogTypes := []string{"INDEX_SLOW_LOGS", "SEARCH_SLOW_LOGS"}
+								for _, str := range validLogTypes {
+									if value == str {
+										return
+									}
+								}
+								errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validLogTypes, value))
+								return
+							},
+						},
+						"cloud_watch_logs_log_group_arn": {
```

Comment:
Nitpick: Can you make this `cloudwatch_log_group_arn`?

We use `cloudwatch` (written together) in other resources and [`aws_cloudwatch_log_group`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_log_group.html) is a resource name too. I'm aware we'll drift from the API here, but I think it's 👌  to make an exception as the name they use there is really odd.